include/alpaka/core/Cuda.hpp

```diff
@@ -177,15 +177,31 @@ namespace alpaka
                 int const & line)
             -> void
             {
-                if(error != CUDA_SUCCESS)
-                {
-                    std::string const sError(std::string(file) + "(" + std::to_string(line) + ") " + std::string(desc) + " : '" + cudaGetErrorName(static_cast<cudaError_t>(error)) +  "': '" + std::string(cudaGetErrorString(static_cast<cudaError_t>(error))) + "'!");
+                if(error == CUDA_SUCCESS)
+                    return;
+
+                char const * cu_err_string = nullptr;
+                char const * cu_err_name = nullptr;
+                CUresult cu_result_name = cuGetErrorName(error, &cu_err_name);
+                CUresult cu_result_string = cuGetErrorString(error, &cu_err_string);
+                std::string sError;
+                if( cu_result_name == CUDA_ERROR_INVALID_VALUE
+                    || cu_result_string == CUDA_ERROR_INVALID_VALUE ) {
```

Comment:
Could it be useful to add and check the two CUDA string separately? Then we could print something even when one of the two methods returns an error. At least I do not see a reason to make check them in combination.